### ISSUE TITLE
False positive E0602 on Python 3.10+ for postponed evaluation of annotations

### ISSUE THREAD

**Issue by ShadowLNC** [PyCQA/pylint#4490]
Similar to #2847 and #2552 but not the same due to changes in Python 3.10.

### Steps to reproduce

repro.py:
```
class X:
    def func(self) -> X:
        pass
```

### Current behavior

```
pylint .\repro.py
************* Module repro
repro.py:1:0: C0114: Missing module docstring (missing-module-docstring)
repro.py:1:0: C0103: Class name "X" doesn't conform to PascalCase naming style (invalid-name)
repro.py:1:0: C0115: Missing class docstring (missing-class-docstring)
repro.py:2:4: C0116: Missing function or method docstring (missing-function-docstring)
repro.py:2:22: E0602: Undefined variable 'X' (undefined-variable)
repro.py:1:0: R0903: Too few public methods (1/2) (too-few-public-methods)
```

### Expected behavior

E0602 should not be raised as postponed evaluation of annotations are enabled by default since Python 3.10.

It is (correctly) not raised if `from __future__ import annotations` is specified.

### Versions

Result of `pylint --version` output:

```
pylint --version
pylint 2.8.2
astroid 2.5.6
Python 3.10.0b1 (tags/v3.10.0b1:ba42175, May  3 2021, 20:22:30) [MSC v.1928 64 bit (AMD64)]
```

Also tried on the prerelease with the same behaviour:

```
pylint --version
pylint 3.0.0a3
astroid 2.5.6
Python 3.10.0b1 (tags/v3.10.0b1:ba42175, May  3 2021, 20:22:30) [MSC v.1928 64 bit (AMD64)]
```

**Comment by ShadowLNC** [comment]
Apologies - looks like this has been deferred to 3.11

**Comment by Pierre-Sassoulas** [comment]
We'll have to handle 3.11 at some point though :)
